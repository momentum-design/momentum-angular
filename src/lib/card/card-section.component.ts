import { Component, Input} from '@angular/core';

@Component({
  selector: 'md-card-section',
  template: `
    <ng-content></ng-content>
  `,
  host: {
    'class': 'md-card-section',
    '[class.md-card-section--full]': 'full',

  }
})
export class CardSectionComponent {
  @Input() full: boolean;

 }
