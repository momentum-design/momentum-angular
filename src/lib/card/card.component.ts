import { Component, ElementRef, Input } from '@angular/core';

@Component({
  selector: 'md-card',
  template: `
    <ng-content></ng-content>
  `,
  styles: [],
  host: {
    'class': 'md-card',
  }
})

export class CardComponent {

}
