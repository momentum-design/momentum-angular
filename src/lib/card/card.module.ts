import { NgModule, CUSTOM_ELEMENTS_SCHEMA, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CardComponent } from './card.component';
import { CardSectionComponent } from './card-section.component';


@NgModule({
  declarations: [CardComponent, CardSectionComponent ],
  imports: [
    CommonModule
  ],
  exports: [CardComponent, CardSectionComponent],
  schemas: [CUSTOM_ELEMENTS_SCHEMA]
})
export class CardModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: CardModule,
      providers: [],
    };
  }
 }
