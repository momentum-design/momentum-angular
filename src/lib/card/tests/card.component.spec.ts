import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { CardModule } from '../card.module';
import { Component } from '@angular/core';
import { CardComponent } from '../card.component';
import { By } from '@angular/platform-browser';
import { dispatchKeyboardEvent } from '../../../../test/utils';
import { SPACE, ENTER } from '@angular/cdk/keycodes';

@Component({
  selector: 'test-card',
  template: `<md-card class="small-3 columns" clickable="true" (cardClicked)="onClick(event);">
  <md-card-section>
      <div>
        <h4>Title</h4>
        <h6>Subtitle</h6>
    </div>
  </md-card-section>
  <md-card-section [full]='true'>
    <img src="https://place-hold.it/400x400">
  </md-card-section>
  <md-card-section>
    Content
  </md-card-section>
  <md-card-section>FOOTER</md-card-section>
</md-card>`
})
class TestCardComponent {
  onClick(event: Event) {}
}

describe('Card Test', () => {
  let component: TestCardComponent;
  let fixture: ComponentFixture<TestCardComponent>;
  let cardElement: HTMLElement;




  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ TestCardComponent ],
      imports: [CardModule]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestCardComponent);
    component = fixture.componentInstance;
    cardElement = fixture.debugElement.nativeElement.querySelector('md-card');
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(cardElement).toBeTruthy();
    expect(cardElement).toMatchSnapshot();
  });

  it('should call the function when clicked', () => {
    spyOn(fixture.componentInstance, 'onClick');
    fixture.detectChanges();
    const cardTestComponent = fixture.debugElement.query(By.directive(CardComponent));
    cardTestComponent.triggerEventHandler('click', null);
    expect(fixture.componentInstance.onClick).toHaveBeenCalled();
  });

  it('should call the function when enter is pressed', () => {
    spyOn(fixture.componentInstance, 'onClick');
    fixture.detectChanges();
    const cardTestComponent = fixture.debugElement.query(By.directive(CardComponent));
    cardTestComponent.triggerEventHandler( 'keydown', {key: 'Enter'});
    expect(fixture.componentInstance.onClick).toHaveBeenCalled();
  });


});
