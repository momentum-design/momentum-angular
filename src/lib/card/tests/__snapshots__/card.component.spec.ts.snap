// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test should create 1`] = `
<md-card
  class="small-3 columns md-card"
>
  <md-card-section
    class="md-card-section"
  >
    <div>
      <h4>
        Title
      </h4>
      <h6>
        Subtitle
      </h6>
    </div>
  </md-card-section>
  <md-card-section
    class="md-card-section md-card-section--full"
    ng-reflect-full="true"
  >
    <img
      src="http://placekitten.com/600/300"
    />
  </md-card-section>
  <md-card-section
    class="md-card-section"
  >
     Content 
  </md-card-section>
  <md-card-section
    class="md-card-section"
  >
    FOOTER
  </md-card-section>
</md-card>
`;
